onnxruntime/core/providers/cpu/ml/label_encoder.h

```diff
@@ -53,57 +54,181 @@ class LabelEncoder_2 final : public OpKernel {
     std::vector<TKey> keys;
     std::vector<TValue> values;
 
-    ORT_THROW_IF_ERROR(info.GetAttrs<TKey>(_key_field_name, keys));
-    ORT_THROW_IF_ERROR(info.GetAttrs<TValue>(_value_field_name, values));
+    ORT_THROW_IF_ERROR(info.GetAttrs<TKey>(key_field_name_, keys));
+    ORT_THROW_IF_ERROR(info.GetAttrs<TValue>(value_field_name_, values));
 
     auto num_keys = keys.size();
     auto num_values = values.size();
-    ORT_ENFORCE(num_keys == num_values,
-                "The ", _key_field_name, " and ", _value_field_name, " attribtues in LabelEncoder ",
-                "(name: ", info.node().Name(), ") must have the same length. ",
-                "However, the number of key is ", num_keys, " and the number of ",
-                "values is ", num_values, ".");
-    _map.reserve(num_keys);
-    for (size_t i = 0; i < num_keys; ++i)
-      _map.emplace(keys[i], values[i]);
+    ORT_ENFORCE(num_keys == num_values, "The ", key_field_name_, " and ", value_field_name_,
+                " attributes in LabelEncoder ", "(name: ", info.node().Name(), ") must have the same length. ",
+                "However, the number of key is ", num_keys, " and the number of ", "values is ", num_values, ".");
+    map_.reserve(num_keys);
+    for (size_t i = 0; i < num_keys; ++i) map_.emplace(keys[i], values[i]);
   }
 
   Status Compute(OpKernelContext* context) const override {
-    const auto* tensor_pointer = context->Input<Tensor>(0);
-    if (tensor_pointer == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
-    const Tensor& X = *tensor_pointer;
-    const TensorShape& shape = X.Shape();
-    Tensor& Y = *context->Output(0, shape);
-
-    auto input = X.template DataAsSpan<TKey>();
-    auto output = Y.template MutableDataAsSpan<TValue>();
-
-    for (int64_t i = 0; i < shape.Size(); ++i) {
-      const auto found = _map.find(input[onnxruntime::narrow<size_t>(i)]);
-      if (found == _map.end())
-        output[onnxruntime::narrow<size_t>(i)] = _default_value;
-      else
-        output[onnxruntime::narrow<size_t>(i)] = found->second;
+    const auto* X = context->Input<Tensor>(0);
+    const TensorShape& shape = X->Shape();
+    auto* Y = context->Output(0, shape);
+
+    auto input = X->template DataAsSpan<TKey>();
+    auto output = Y->template MutableDataAsSpan<TValue>();
+    auto input_iter = input.begin();
+    auto output_iter = output.begin();
+    while (input_iter != input.end()) {
+      const auto found = map_.find(*input_iter);
+      *output_iter = found == map_.end() ? default_value_ : found->second;
+      output_iter++;
+      input_iter++;
```

Comment:
Last comment, let's make it pre-increment. Takes place for every tensor element may subtly affect perf, which may even not show on a radar.